python/pyiceberg/cli/output.py

```diff
@@ -14,9 +14,11 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+from __future__ import annotations
```

Comment:
I like Optional[str] too, but I think for some reason when I run `make lint` locally it turns it into the later format i.e. `str | None ` Do we need to tweak the linter config to enforce one format over the other?

Reply:
This is because the import statement. PyUpgrade will automatically rewrite the imports: https://github.com/asottile/pyupgrade#pep-604-typing-rewrites We can change this in another PR.

Reply:
I've reverted to not use `from __future__ import annotations` in a recent commit.

Reply:
It should have been fixed now https://github.com/apache/iceberg/pull/6114 is in. Delayed loading of the annotations (that's what the import does), can be helpful sometime :)